src/settings/components/RoleDetails/AccordionUsers.js

```diff
@@ -53,11 +53,11 @@ const AccordionUsers = ({ roleId }) => {
       }
       displayWhenClosed={
         <Badge>
-          {usersData?.length || 0}
+          {users?.length || 0}
         </Badge>
       }
       displayWhenOpen={<AssignUsers
-        selectedUsers={usersData}
+        selectedUsers={users}
```

Comment:
The issue with this ticket is that the wrong variable was accidentally provided. Instead of using `users` which is array of object that includes `id`, the variable usersData was provided as [] { fullName, patronGroup }.